google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/sql/ExecuteQueryCallableTest.java

```diff
@@ -108,13 +114,54 @@ public void testExecuteQueryRequestsAreNotRetried() {
     assertThat(fakeService.attempts).isEqualTo(1);
   }
 
+  @Test
+  public void testExecuteQueryRequestsSetDefaultDeadline() throws IOException {
+    SqlServerStream stream =
+        stub.executeQueryCallable().call(Statement.of("SELECT * FROM table"));
+    Iterator<SqlRow> iterator = stream.rows().iterator();
+    // We don't care about this but are reusing the fake service that tests retries
+    assertThrows(UnavailableException.class, iterator::next).getCause();
+    // We have 30s default, we assume less than 1s has been burned when the fake service sets it
+    assertThat(fakeService.deadlineMillisRemaining).isIn(Range.closed(29000L, 30100L));
+  }
+
+  @Test
+  public void testExecuteQueryRequestsRespectOverridenDeadline() throws IOException {
+    BigtableDataSettings.Builder overrideSettings =
+        BigtableDataSettings.newBuilderForEmulator(server.getPort())
+            .setProjectId("fake-project")
+            .setInstanceId("fake-instance");
+    overrideSettings
```

Comment:
Can we also make the operation retryable and make sure it's not getting retried?